### PR TITLE
Add max button to amount field

### LINKED ISSUE #127: Add max button on amount input

### PR DESCRIPTION
**Pull request by jubalm**
Resolves #127 

- [x] add max button
- [x] add loading indicator if balance is busy

| Screenshot 2023-06-04 at 10 05 43 AM | 
|:--:| 
| *Idle State* |

| Screenshot 2023-06-04 at 10 05 47 AM | 
|:--:| 
| *Busy State* |

### CODE REVIEW

**Review by 0xjimmy — APPROVED** [review]
Would be good to
 - show the users token balance, maybe have it inside the asset selector card? Or next to the amount input
Could be one of these places perhaps. 
image



- show an error + input red/warning state when the amount input is > than the users balance.

File: app/ts/components/AmountField.tsx [hunk 1]
```diff
@@ -38,18 +44,31 @@ export const AmountField = (props: Props) => {
 		}
 	}
 
+	const isCalculatingMax = useComputed(() => tokenBalance.value.state === 'pending')
+
+	const handleMax = () => {
```
Comment (KillariDev) [inline app/ts/components/AmountField.tsx:49]: maybe some better function name than `handle` as its quite vague on what it does? I am not sure what this function does. I guess it updates the token balance to the max?

File: app/ts/components/TransferPage/index.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ const MainPanel = () => {
 	const { transaction, data, send, clearData } = useTransfer()
 
 	const handleTokenSelect = (token?: TokenMeta) => {
```
Comment (KillariDev) [inline app/ts/components/TransferPage/index.tsx:35]: maybe better function name than `handle` This could be `updateTokenSelect` instead?

**Review by KillariDev — APPROVED** [review]
seems like there's some conflicts too

**Comment by MicahZoltu** [comment]
> * show the users token balance, maybe have it inside the asset selector card? Or next to the amount input

The user's balance is there, it is just in a light grey that is hard to see.  I don't think we should worry about fixing the coloring in this PR though, we can deal with that in a future PR.

**Comment by jubalm** [comment]
> Would be good to
> 
> * show the users token balance, maybe have it inside the asset selector card? Or next to the amount input
>   Could be one of these places perhaps.
>   image
> * show an error + input red/warning state when the amount input is > than the users balance.

created separate issue for this feature #138

File: app/ts/components/AmountField.tsx (hunk 1 above)
Comment (jubalm) [reply app/ts/components/AmountField.tsx]: changed to `setTokenBalanceAsAmount` to describe what it does.

File: app/ts/components/TransferPage/index.tsx (hunk 2 above)
Comment (jubalm) [reply app/ts/components/TransferPage/index.tsx]: changed to `setUserSelectedToken` to be more descriptive in commit [e185417](https://github.com/DarkFlorist/lunaria/pull/133/commits/e18541702e4d255983547a280bb996cf4762e738)

**Comment by jubalm** [comment]
> seems like there's some conflicts too

the conflict was a previously merged css change only, rebasing to master fixes this branch.

**Pull request merged**